src/test/java/com/puppycrawl/tools/checkstyle/SuppressionsStringPrinterTest.java

```diff
@@ -0,0 +1,129 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2018 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle;
+
+import static com.puppycrawl.tools.checkstyle.internal.utils.TestUtil.isUtilsClassHasPrivateConstructor;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import antlr.NoViableAltException;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+
+public class SuppressionsStringPrinterTest extends AbstractTreeTestSupport {
+
+    private static final String LINE_SEPARATOR = System.getProperty("line.separator");
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/suppressionsstringprinter";
+    }
+
+    @Test
+    public void testIsProperUtilsClass() throws ReflectiveOperationException {
+        assertTrue("Constructor is not private",
+                isUtilsClassHasPrivateConstructor(SuppressionsStringPrinter.class, true));
+    }
+
+    @Test
+    public void testCorrect() throws Exception {
+        final String expected = new String(Files.readAllBytes(Paths.get(
+                getPath("ExpectedSuppressionsStringPrinter.txt"))),
+                StandardCharsets.UTF_8).replaceAll("\\\\r\\\\n", "\\\\n");
```

Comment:
such code mean that after clone of repo to windows, all files become `\r\n` (expected, as it is natural for windows).
Replacement to unix style mean that code that generate output does not consider OS difference, so output on windows will be damaged in commanline.
Please recheck builder of output for using  OS specific line-ending.

Reply:
@romani 
> Replacement to unix style mean that code that generate output does not consider OS difference,

We do this for alot of tests like this. See `AbstractPathTestSupport` and `MainTest`. I prefer we keep 1 file and not duplicate the file between each OS line separator.
If we inline the files to strings, this won't be an issue since we can use System's line separator.
If this remains as a file, we should use `AbstractPathTestSupport.readFile` and not duplicate the code.

Reply:
Yes, but let's wait for feedback from @timurt .